### PR TITLE
Fixed issue of not requiring SAS permission for some specific operations

### PR DESCRIPTION
**Pull request by EmmaZhu**
Thanks for contribution! Please go through following checklist before sending PR.

### PR Branch Destination

- For Azurite V3, please send PR to `main` branch.
- For legacy Azurite V2, please send PR to `legacy-dev` branch.

### Always Add Test Cases

Make sure test cases are added to cover the code change.

### Add Change Log

Add change log for the code change in `Upcoming Release` section in `ChangeLog.md`.

### Development Guideline

Please go to CONTRIBUTION.md for steps about setting up development environment and recommended Visual Studio Code extensions.

### CODE REVIEW

File: src/blob/authentication/OperationAccountSASPermission.ts [hunk 1]
```diff
@@ -77,16 +75,16 @@ OPERATION_ACCOUNT_SAS_PERMISSIONS.set(
...
-      AccountSASResourceType.Container +
-      AccountSASResourceType.Object,
+    AccountSASResourceType.Container +
+    AccountSASResourceType.Object,
+    AccountSASPermission.Read +
+    AccountSASPermission.Create +
+    AccountSASPermission.Delete +
+    AccountSASPermission.List +
+    AccountSASPermission.Process +
     AccountSASPermission.Read +
```
Comment (blueww) [inline src/blob/authentication/OperationAccountSASPermission.ts:85]: It looks there are dup AccountSASPermission.Read .

File: src/blob/authentication/OperationAccountSASPermission.ts [hunk 2]
```diff
@@ -35,33 +35,31 @@ export class OperationAccountSASPermission {
...
-    if (this.permission) {
-      for (const p of this.permission) {
-        if (permissions.toString().includes(p)) {
-          return true;
-        }
-      }
-      return false;
-    }
-    else {
+    if (this.permission === AccountSASPermission.Any) {
```
Comment (blueww) [inline src/blob/authentication/OperationAccountSASPermission.ts:53]: Please Add comment this is only for blob batch.

File: src/blob/authentication/OperationAccountSASPermission.ts [hunk 3]
```diff
@@ -35,33 +35,31 @@ export class OperationAccountSASPermission {
...
-    if (this.resourceType) {
-      for (const p of this.resourceType) {
-        if (resourceTypes.toString().includes(p)) {
-          return true;
-        }
-      }
-      return false;
-    }
-    else {
+    if (this.resourceType === AccountSASResourceType.Any) {
```
Comment (blueww) [inline src/blob/authentication/OperationAccountSASPermission.ts:38]: Please Add comment this is only for blob batch.

File: src/blob/authentication/BlobSASPermissions.ts [hunk 4]
```diff
@@ -4,10 +4,11 @@ export enum BlobSASPermission {
...
-  execute  = "e",
-  SetImmutabilityPolicy  = "i",
-  permanentDelete  = "y"
+  DeleteVersion = "x",
+  Tag = "t",
+  Move = "m",
+  execute = "e",
+  SetImmutabilityPolicy = "i",
+  permanentDelete = "y",
+  Any = "AnyPermission"
```
Comment (blueww) [inline src/blob/authentication/BlobSASPermissions.ts:13]: Please Add comment this is only for blob batch.

File: src/blob/authentication/ContainerSASPermissions.ts [hunk 5]
```diff
@@ -4,5 +4,6 @@ export enum ContainerSASPermission {
   Create = "c",
   Write = "w",
   Delete = "d",
-  List = "l"
+  List = "l",
+  Any = "AnyPermission"
```
Comment (blueww) [inline src/blob/authentication/ContainerSASPermissions.ts:8]: Please Add comment this is only for blob batch.

File: src/blob/authentication/OperationAccountSASPermission.ts [hunk 6]
```diff
@@ -95,16 +93,16 @@ OPERATION_ACCOUNT_SAS_PERMISSIONS.set(
...
-      AccountSASResourceType.Container +
-      AccountSASResourceType.Object,
+    AccountSASResourceType.Container +
+    AccountSASResourceType.Object,
+    AccountSASPermission.Read +
+    AccountSASPermission.Create +
+    AccountSASPermission.Delete +
+    AccountSASPermission.List +
+    AccountSASPermission.Process +
     AccountSASPermission.Read +
```
Comment (blueww) [inline src/blob/authentication/OperationAccountSASPermission.ts:103]: dup read?

File: src/blob/authentication/OperationBlobSASPermission.ts [hunk 7]
```diff
@@ -3,24 +3,23 @@ import { BlobSASPermission } from "./BlobSASPermissions";
...
-    if (this.permission !== "") {
-      for (const p of this.permission) {
-        if (permissions.toString().includes(p)) {
-          return true;
-        }
-      }
-      return false;
-    }
-    else {
+    if (this.permission === ContainerSASPermission.Any) {
```
Comment (blueww) [inline src/blob/authentication/OperationBlobSASPermission.ts:13]: Please comment this is only for blob batch.

File: src/blob/authentication/OperationAccountSASPermission.ts (hunk 1 above)
Comment (EmmaZhu) [reply src/blob/authentication/OperationAccountSASPermission.ts:87]: It's there before. Only prettier made some  indent changes.

File: src/blob/authentication/OperationAccountSASPermission.ts (hunk 6 above)
Comment (EmmaZhu) [reply src/blob/authentication/OperationAccountSASPermission.ts:105]: It's there before. Only prettier made some indent changes.

File: tests/blob/sas.test.ts [hunk 8]
```diff
@@ -708,6 +708,48 @@ describe("Shared Access Signature (SAS) authentication", () => {
     await containerClient.delete();
   });
 
+  it("GetProperties on container should fail with container SAS @loki @sql", async () => {
```
Comment (blueww) [inline tests/blob/sas.test.ts:711]: Please also test blob sas, account sas, and should also test on one of service API, and might(optional) can also add create container.

**Review by blueww — APPROVED** [review]

**Pull request merged**